x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -546,12 +456,12 @@ export class Authenticator {
    * @param sessionValue Current session value.
    */
   private *providerIterator(
-    sessionValue: ProviderSession | null
+    sessionValue: SessionValue | null
   ): IterableIterator<[string, BaseAuthenticationProvider]> {
     // If there is no session to predict which provider to use first, let's use the order
     // providers are configured in. Otherwise return provider that owns session first, and only then the rest
     // of providers.
-    if (!sessionValue) {
+    if (!sessionValue || !this.providers.has(sessionValue.provider.name)) {
```

Comment:
__note:__ this case is very-very rare that why we never heard of it. Found it while writing new tests - it's better to fix.